packager/VisibilityChecker.cc

```diff
@@ -456,7 +456,7 @@ class VisibilityCheckerPass final {
             if (auto e = ctx.beginError(lit.loc, core::errors::Packager::MissingImport)) {
                 auto &pkg = ctx.state.packageDB().getPackageInfo(otherPackage);
                 e.setHeader("`{}` resolves but its package is not imported", lit.symbol.show(ctx));
-                bool isTestImport = otherFile.data(ctx).isPackagedTest();
+                bool isTestImport = otherFile.data(ctx).isPackagedTest() || ctx.file.data(ctx).isPackagedTest();
```

Comment:
Just to walk through my understanding here:

1. If `otherFile` is from a test package, then (almost, see below) obviously this should be a `test_import`.  Fine.
2. If `otherFile` is from a regular package, and we are from a test package, then `otherFile`'s package should be a `test_import`.  OK.
3. If `otherFile` is from a regular package, and we are from a regular package, then a plain `import` is called for.  Fine.

I think it's possible that we can wind up in a weird situation here, because there's a check above:

```c++
        // If the imported symbol comes from the test namespace, we must also be in the test namespace.
        if (otherFile.data(ctx).isPackagedTest() && !this->insideTestFile) {
            if (auto e = ctx.beginError(lit.loc, core::errors::Packager::UsedTestOnlyName)) {
                e.setHeader("`{}` is defined in a test namespace and cannot be referenced in a non-test file",
                            lit.symbol.show(ctx));
            }
        }
```

that doesn't return from this method.  So it is possible to get to this autocorrect with `otherFile` from a test package and us *not* from a test package.  I guess in that case we should not be suggesting an autocorrect and all?  Worth handling?  Maybe, but probably in a follow up.

Reply:
It was a one line change and I'm already changing that CLI test in this PR so I figured it'd be easier to change now.